File: SymmetricTree/isSymmetric.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSymmetric(TreeNode root) {
        return root == null || helper(root.left, root.right); 
    }
    public boolean helper(TreeNode lt, TreeNode rt) {
        if(lt == null || rt == null)
            return lt == rt;
        if(lt.val == rt.val)
            return helper(lt.left, rt.right) && helper(lt.right, rt.left);
        else 
            return false;
    }
}

File: SymmetricTree/isSymmetric.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    if(root == null) return true;
    return helper(root.left, root.right);
};
var helper = function(lt, rt) {
    if(lt == null && rt == null) return true;
    if(lt == null || rt == null) return false;
    if(lt.val == rt.val) {
        return helper(lt.left, rt.right) && helper(lt.right, rt.left);
    } else 
        return false;
}